dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/TasksRunner.java

```diff
@@ -70,18 +72,28 @@ public TasksRunner(
   }
 
   private TaskRunContext createContext(
-      OutputHandleFactory sinkFactory, Handle handle, int threadPoolSize, Impl state) {
-    return new TaskRunContext(sinkFactory, handle, threadPoolSize) {
-      @Override
-      public TaskState getTaskState(Task<?> task) {
-        return state.getTaskState(task);
-      }
-
-      @Override
-      public <T> T runChildTask(Task<T> task) throws MetadataDumperUsageException {
-        return handleTask(task);
-      }
-    };
+      OutputHandleFactory sinkFactory,
+      Handle handle,
+      int threadPoolSize,
+      Impl state,
```

Comment:
The `Impl` here can be replaced with the interface `TaskSetState`, for which it is an implementation class.

Optionally, the same can be done for the constructor, but this requires changing the `state` attribute as well.

Reply:
The setters are not available in the interface, therefore `Impl` is needed in the `TasksRunner` class. I have removed this parameter from `createContext` method though, because it turns out that after making the changes from your previous comment, it is no longer needed here.